tests/testthat/test-intensity_levels.R

```diff
@@ -0,0 +1,66 @@
+test_that("Test that input argument checks work", {
+
+  start_date <- as.Date("2021-01-04")
+  end_date <- as.Date("2023-12-31")
+
+  weekly_dates <- seq.Date(from = start_date,
+                           to = end_date,
+                           by = "week")
+
+  obs <- rpois(length(weekly_dates), 1000)
+
+  tsd_data <- tsd(
+    observed = obs,
+    time = as.Date(weekly_dates),
+    time_interval = "week"
+  )
+
+  expect_no_error(intensity_levels(tsd_data))
+
+  expect_error(intensity_levels(tsd_data, decay_factor = 3))
+
+  expect_error(intensity_levels(tsd_data, n_peak = c(1, 2)))
+
+  expect_error(intensity_levels(tsd_data, disease_threshold = "hey"))
+
+  expect_error(intensity_levels(tsd_data, season_weeks = c(2, 10)))
```

Comment:
Ideally, you would check the error is the one that you expect. 
That is, use the "regex" argument of "expect_error" to compare error messages.